app/pages/project/vpcs/VpcPage/tabs/VpcGatewaysTab.tsx

```diff
@@ -5,3 +5,64 @@
  *
  * Copyright Oxide Computer Company
  */
+
+import { createColumnHelper } from '@tanstack/react-table'
+import { useMemo } from 'react'
+import { Outlet, type LoaderFunctionArgs } from 'react-router-dom'
+
+import { apiQueryClient, type InternetGateway } from '~/api'
+import { getVpcSelector, useVpcSelector } from '~/hooks/use-params'
+import { makeLinkCell } from '~/table/cells/LinkCell'
+import { Columns } from '~/table/columns/common'
+import { useQueryTable } from '~/table/QueryTable'
+import { EmptyMessage } from '~/ui/lib/EmptyMessage'
+import { ALL_ISH } from '~/util/consts'
+import { pb } from '~/util/path-builder'
+
+const colHelper = createColumnHelper<InternetGateway>()
+
+VpcInternetGatewaysTab.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project, vpc } = getVpcSelector(params)
+  const query = { project, vpc, limit: ALL_ISH }
+  await Promise.all([
+    apiQueryClient.prefetchQuery('internetGatewayList', { query }),
+    apiQueryClient.prefetchQuery('internetGatewayIpAddressList', { query }),
+    apiQueryClient.prefetchQuery('internetGatewayIpPoolList', { query }),
```

Comment:
While merging the `useQueryTable` change in here I noticed the `gateway` query param is not required on these two list endpoints, which is why they can be called here without a type error despite the fact that there is no gateway in scope. This turns out to be a bug in the OpenAPI schema generation — the gateway param is required in the Rust code but not in the OpenAPI schema.

https://github.com/oxidecomputer/omicron/issues/7155

This isn't really a problem for us, we just have to make sure to call it correctly, i.e., with the gateway. Here we can just remove the calls. On the gateway page you have it right.